cmd/chisel/cmd_cut_test.go

```diff
@@ -0,0 +1,419 @@
+{"kind":"path","path":"/foo","mode":"0644","slices":["other-package_otherslice"],"sha256":"2c26b46b68ffc68ff99b453c1d30413413422d706483bfa0f98a5e886266e7ae","size":3}
+{"kind":"slice","name":"other-package_otherslice"}
+{"kind":"slice","name":"test-package_manifest"}
+{"kind":"slice","name":"test-package_myslice"}
+`,
+}, {
+	summary: "No DB if corresponding slice(s) are not selected",
+	release: map[string]string{
+		"slices/mydir/test-package.yaml": `
+			package: test-package
+			slices:
+				myslice:
+					contents:
+						/dir/file:
+						/dir/file-copy:  {copy: /dir/file}
+						/other-dir/file: {symlink: ../dir/file}
+						/dir/text-file:  {text: data1}
+						/dir/foo/bar/:   {make: true, mode: 01777}
+				manifest:
+					contents:
+						/db/**: {generate: manifest}
+		`,
+	},
+	slices: []string{"test-package_myslice"},
+	filesystem: map[string]string{
+		"/dir/":           "dir 0755",
+		"/dir/file":       "file 0644 cc55e2ec",
+		"/dir/file-copy":  "file 0644 cc55e2ec",
+		"/dir/foo/":       "dir 0755",
+		"/dir/foo/bar/":   "dir 01777",
+		"/dir/text-file":  "file 0644 5b41362b",
+		"/other-dir/":     "dir 0755",
+		"/other-dir/file": "symlink ../dir/file",
+	},
+}}
+
+var defaultChiselYaml = `
+	format: v1
+	archives:
+		ubuntu:
+			version: 22.04
+			components: [main, universe]
+			public-keys: [test-key]
+	public-keys:
+		test-key:
+			id: ` + testKey.ID + `
+			armor: |` + "\n" + testutil.PrefixEachLine(testKey.PubKeyArmor, "\t\t\t\t\t\t") + `
+`
+
+func (s *ChiselSuite) TestCut(c *C) {
+	for _, test := range cutTests {
+		c.Logf("Summary: %s", test.summary)
+
+		if _, ok := test.release["chisel.yaml"]; !ok {
+			test.release["chisel.yaml"] = string(defaultChiselYaml)
+		}
+
+		if test.pkgs == nil {
+			test.pkgs = map[string][]byte{
+				"test-package": testutil.PackageData["test-package"],
+			}
+		}
+		archivePackages = test.pkgs
+
+		releaseDir := c.MkDir()
+		for path, data := range test.release {
+			fpath := filepath.Join(releaseDir, path)
+			err := os.MkdirAll(filepath.Dir(fpath), 0755)
+			c.Assert(err, IsNil)
+			err = os.WriteFile(fpath, testutil.Reindent(data), 0644)
+			c.Assert(err, IsNil)
+		}
+
+		restore := fakeOpenArchive()
+		defer restore()
+
+		targetDir := c.MkDir()
+		args := []string{"cut", "--release", releaseDir + "/", "--root", targetDir + "/"}
+		args = append(args, test.slices...)
+
+		extra, err := chisel.Parser().ParseArgs(args)
+		if test.err != "" {
+			c.Assert(err, ErrorMatches, test.err)
+			continue
+		}
+		c.Assert(err, IsNil)
+		c.Assert(len(extra), Equals, 0)
+
+		if test.filesystem != nil {
+			c.Assert(testutil.TreeDump(targetDir), DeepEquals, test.filesystem)
+		}
+
+		test.db = strings.TrimLeft(test.db, "\n")
+		if test.dbPaths != nil {
+			for _, path := range test.dbPaths {
+				db, err := readZSTDFile(filepath.Join(targetDir, path))
+				c.Assert(err, IsNil)
+				c.Assert(db, DeepEquals, test.db)
+			}
+		}
+	}
+}
+
+var archivePackages map[string][]byte
```

Comment:
Can we avoid having this global state? Right now the tests run sequentially but, is possible, it is better for them to be independent.

Just adding the packages as the argument to `fakeOpenArchive` should be enough.

Reply:
Done, thanks! 33d42c4